src/core/public/application/application_service.test.ts

```diff
@@ -726,6 +726,47 @@ describe('#start()', () => {
       `);
     });
 
+    it('updates httpLoadingCount$ while mounting', async () => {
+      // Use a memory history so that mounting the component will work
+      const { createMemoryHistory } = jest.requireActual('history');
+      const history = createMemoryHistory();
+      setupDeps.history = history;
+
+      // Create some dummy applications
+      const { register } = service.setup(setupDeps);
+      register(Symbol(), createApp({ id: 'alpha' }));
+      register(Symbol(), createApp({ id: 'beta' }));
+      register(Symbol(), createApp({ id: 'gamma' }));
+      register(Symbol(), createApp({ id: 'delta' }));
+
+      const { navigateToApp, getComponent } = await service.start(startDeps);
+      const httpLoadingCount$ = startDeps.http.addLoadingCountSource.mock.calls[0][0];
+      const stop$ = new Subject();
+      const promise = httpLoadingCount$.pipe(bufferCount(8), takeUntil(stop$)).toPromise();
+      mount(getComponent()!);
+
+      await navigateToApp('alpha');
+      await navigateToApp('beta');
+      await navigateToApp('gamma');
+      await navigateToApp('delta');
+      stop$.next();
+
+      const loadingCounts = await promise;
+
+      expect(loadingCounts).toMatchInlineSnapshot(`
+        Array [
+          0,
+          1,
+          0,
```

Comment:
NIT: We are not really testing that the count is `1` while awaiting the mount function, and `0` before/after. However I don't see how to properly assert that in an UT as it requires the mount function to be executed. 

We could add a test for that in `src/core/public/application/integration_tests/application_service.test.tsx`
but maybe the added coverage in `app_container.test.tsx` is sufficient.

Reply:
Was able to create a test here where I can control when the application completes mounting in order to execute. an assertion at the right time.